Test/UnitTest1.cs

```diff
@@ -1,8 +1,32 @@
-namespace Test;
+using Training;
 
-[TestClass]
-public class UnitTest1 {
-   [TestMethod]
-   public void TestMethod1 () {
+namespace Test {
+   [TestClass]
+   public class UnitTest1 {
+      [TestMethod]
+      public void EnqueueTest () {
+         mQueue.Enqueue (1);
+         mQueue.Enqueue (2);
+         Assert.AreEqual (2, mQueue.Count);
+      }
+
+      [TestMethod]
+      public void DequeueTest () {
+         Assert.ThrowsException<InvalidOperationException> (() => mQueue.Dequeue ());
+         mQueue.Enqueue (1);
+         mQueue.Enqueue (2);
+         Assert.AreEqual (1, mQueue.Dequeue ());
+         Assert.AreEqual (1, mQueue.Count);
+      }
+
+      [TestMethod]
+      public void PeekTest () {
+         Assert.ThrowsException<InvalidOperationException> (() => mQueue.Peek ());
+         mQueue.Enqueue (1);
+         mQueue.Enqueue (2);
+         Assert.AreEqual (1, mQueue.Peek ());
+      }
+
+      TQueue<int> mQueue = new ();

```

Comment:
'mQueue' is not initialized for each test. Are all tests getting passed?

Reply:
'mQueue' is initialized outside in common and accessed in all test methods. Yes ma'am, all the tests are getting passed.
Should I initialize it for each test or is this ok?